### PR TITLE
runbutton with extra error and warning icon

### PR DESCRIPTION
**Pull request by kasperpeulen**
Looks like this now:

image

image

image

### CODE REVIEW

**Comment by kasperpeulen** [comment]
the yellow one now looks like this:
image

File: lib/editing/editor_codemirror.dart [hunk 1]
```diff
@@ -240,6 +248,22 @@ class _CodeMirrorDocument extends Document {
...
+    if (containsError) {
+      path.attributes["d"] = "M5,3H19A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5A2,2 0 0,1 5,3M13,13V7H11V13H13M13,17V15H11V17H13Z";
+      path.parent.classes.add("error");
+      path.parent.classes.remove("warning");
+
+   } else if (containsWarning) {
+      path.attributes["d"] = "M5,3H19A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5A2,2 0 0,1 5,3M13,13V7H11V13H13M13,17V15H11V17H13Z";
+      path.parent.classes.add("warning");
+      path.parent.classes.remove("error");
+   }
```
Comment (devoncarew) [inline lib/editing/editor_codemirror.dart]: nit: `} else {`

File: lib/editing/editor_codemirror.dart [hunk 2]
```diff
@@ -240,6 +248,22 @@ class _CodeMirrorDocument extends Document {
 ////      cm.setGutterMarker(an.line - 1, _gutterId,
 ////          _makeMarker(an.type, an.message, an.start, an.end));
     }
+    var path = html.querySelector("#runbutton path");
```
Comment (devoncarew) [inline lib/editing/editor_codemirror.dart]: Could you move the run button updating code into it's own method? That'll make sure this `setAnnotations` method doesn't get too long. Some thing like `_updateRunButton({bool hasError: false, bool hasWarning: false})`.

File: web/styles.css [hunk 3]
```diff
@@ -154,6 +154,27 @@ button {
...
+  height: 18px;
+  padding: 0;
+  fill: #45914b;
+}
+
+#runbutton svg.error {
+  padding: 0 0 1px 1px;
+  width: 17px;
+  height: 17px;
+  fill: #d78870;
```
Comment (devoncarew) [inline web/styles.css]: Can we use the same colors for red and yellow that we do for the error underlines? I think that's `#F7977A` and `#FFF79A`.

File: lib/editing/editor_codemirror.dart [hunk 4]
```diff
@@ -240,6 +248,21 @@ class _CodeMirrorDocument extends Document {
...
+    if (containsError) {
+      path.attributes["d"] = "M5,3H19A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5A2,2 0 0,1 5,3M13,13V7H11V13H13M13,17V15H11V17H13Z";
+      path.parent.classes.add("error");
+      path.parent.classes.remove("warning");
+
+   } else if (containsWarning) {
+      path.attributes["d"] = "M5,3H19A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5A2,2 0 0,1 5,3M13,13V7H11V13H13M13,17V15H11V17H13Z";
```
Comment (devoncarew) [inline lib/editing/editor_codemirror.dart]: It looks like this path / avg icon is the same as the error. Perhaps pull this out into a final local contant in the method?

**Comment by devoncarew** [comment]
Thanks for the PR! The UI for the run button looks nice; I'm much happier w/ this look than I was with some of the previous ones under discussion.

File: lib/editing/editor_codemirror.dart (hunk 4 above)
Comment (kasperpeulen) [inline lib/editing/editor_codemirror.dart]: @devoncarew  I'm fine with doing that, that is what I started with. But after some time using, it felt to me that it is a little bit too bright or something compared with the grey of the button. 

But if we do this, maybe we should use a little more bright green ?

Comment (devoncarew) [inline lib/editing/editor_codemirror.dart]: @kasperpeulen, let's go with the colors you have now; good to know you did try the others.

**Comment by devoncarew** [comment]
merging...

**Comment by sethladd** [comment]
awesome!

**Pull request merged**